packages/react-vapor/src/components/table-hoc/TableHeaderWithSort.tsx

```diff
@@ -1,91 +1,114 @@
-        this.props.onMount();
-    }
-
-    componentWillUnmount() {
-        this.props.onUnmount();
-    }
-
-    render() {
-        const headerCellClasses = classNames(this.props.className, 'admin-sort', {
-            'admin-sort-ascending': this.props.sorted === true,
-            'admin-sort-descending': this.props.sorted === false,
-        });
-
-        const getSvg = classNames({
-            'sorted-asc': this.props.sorted === true,
-            'sorted-desc': this.props.sorted === false,
-            'asc-desc': this.props.sorted === undefined,
-        });
-
-        if (this.props.isLoading) {
-            return (
-                <th id={this.props.id}>
-                    <TextLoadingPlaceholder small />
-                </th>
-            );
-        }
-
-        return (
-            <th id={this.props.id} className={headerCellClasses} onClick={() => this.props.onSort()}>
-                {this.props.children}
-                <div className="admin-sort-icon">
-                    <Svg svgName={getSvg} className="tables-sort icon" />
-                </div>
-            </th>
-        );
-    }
-}
+import classNames from 'classnames';
+import * as React from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import * as _ from 'underscore';
+
+import {IReactVaporState} from '../../ReactVaporState';
+import {TextLoadingPlaceholder} from '../loading/components/TextLoadingPlaceholder';
+import {Svg} from '../svg/Svg';
+import {TableHeaderActions} from './actions/TableHeaderActions';
+import {ITableWithSortState} from './reducers/TableWithSortReducers';
+
+export interface ITableHeaderWithSortOwnProps {
+    id: string;
+    tableId: string;
+    isLoading?: boolean;
+    isDefault?: boolean;
+    fixWidth?: boolean;
+}
+
+export interface HOCTableHeaderStateProps {
+    sorted: boolean;
+}
+
+export interface ITableHeaderWithSortDispatchProps {
+    onMount: () => void;
+    onSort: () => void;
+    onUnmount: () => void;
+}
+
+export interface ITableHeaderWithSortProps
+    extends ITableHeaderWithSortOwnProps,
+        Partial<HOCTableHeaderStateProps>,
+        Partial<ITableHeaderWithSortDispatchProps> {}
+
+export const TableHeaderWithSort: React.FC<
+    ITableHeaderWithSortProps & React.HTMLAttributes<HTMLTableHeaderCellElement>
+> = ({className, isLoading, id, tableId, isDefault, fixWidth, children}) => {
+    const dispatch = useDispatch();
+    const {sorted} = useSelector((state: IReactVaporState) => {
+        const tableSort: ITableWithSortState = _.findWhere(state.tableHOCHeader, {id});
+
+        return {
+            sorted: tableSort && tableSort.isAsc,
+        };
+    });
+
+    const onMount = () => dispatch(TableHeaderActions.addTableHeader(id, tableId, isDefault));
+    const onSort = () => dispatch(TableHeaderActions.sortTable(id));
+    const onUnmount = () => dispatch(TableHeaderActions.removeTableHeader(id));
+
+    const targetRef = React.useRef<HTMLTableHeaderCellElement>();
```

Comment:
```suggestion
    const cellRef = React.useRef<HTMLTableHeaderCellElement>();
```
I feel targetRef is too generic... it confused me at lines like this one ` width: targetRef.current.offsetWidth,` :sweat_smile:

Reply:
good suggestion